src/pages/index.js

```diff
@@ -15,80 +15,194 @@ import Section from "../components/Section";
+    userInfo.setUserInfo(userData.name, userData.about);
+    userInfo.setUserAvatar(userData.avatar);
+  })
+  .catch((err) => {
+    console.error(`There is an Error: `, err); // log the error to the console
+  });
 
 //Specific Event Listeners
 addNewCardButton.addEventListener("click", () => {
   formValidators["place-form"].toggleButtonState();
   newCardPopup.open();
 });
 
+let userData;
+
 editProfileButton.addEventListener("click", () => {
-  const userInfo = editUserInfo.getUserInfo();
+  userData = userInfo.getUserInfo();
   editUserInfoPopup.setInputValues({
-    title: userInfo.name,
-    description: userInfo.job,
+    title: userData.name,
+    description: userData.job,
   });
   formValidators["profile-form"].resetValidation();
   editUserInfoPopup.open();
 });
 
+editAvatarButton.addEventListener("click", () => {
+  userData = userInfo.getUserInfo();
+  editAvatarPopup.setInputValues({
+    link: userData.src,
+  });
+  formValidators["avatar-form"].toggleButtonState();
+  editAvatarPopup.open();
+});
+
 // Functions
 function handleImageClick(link, name) {
   imagePreviewPopup.open(link, name);
 }
 
 function handleAddCardSubmit(cardData) {
-  cardSection.addItem(
-    createCard({ name: cardData.title, link: cardData.link })
-  );
+  newCardPopup.setButtonText(true);
+  api
+    .addNewCard(cardData.title, cardData.link)
+    .then((data) => {
+      cardSection.addItem(createCard(data));
+    })
+    .catch((err) => {
+      console.error(`There is an Error: `, err); // log the error to the console
+    })
+    .finally(() => {
+      newCardPopup.setButtonText(false);
+    });
+
   newCardPopup.close();
 }
 
 function handleProfileEditSubmit(cardData) {
-  editUserInfo.setUserInfo(cardData.title, cardData.description);
+  editUserInfoPopup.setButtonText(true);
+  api
+    .editUserInfo(cardData.title, cardData.description)
+    .then(() => {
+      userInfo.setUserInfo(cardData.title, cardData.description);
+    })
+    .catch((err) => {
+      console.error(`There is an Error: `, err); // log the error to the console
+    })
+    .finally(() => {
+      editUserInfoPopup.setButtonText(false);
+    });
   editUserInfoPopup.close();
 }
 
+function handleCardDeleteClick(card) {
+  deleteCardPopup.open();
+  deleteCardPopup.setDeleteText(true);
+  deleteCardPopup.setConfirmDelete(() => {
+    api
+      .deleteCard(card._id)
+      .then(() => {
+        card.removeCard();
+        deleteCardPopup.close();
+      })
+      .catch((err) => {
+        console.error(`There is an Error: `, err); // log the error to the console
+      })
+      .finally(() => {
+        deleteCardPopup.setDeleteText(false);
+      });
+  });
+}
+
+function handleAvatarEditSubmit(data) {
+  editAvatarPopup.setButtonText(true);
+  api
+    .updateAvatar(data.link)
+    .then(() => {
+      userInfo.setUserAvatar(data.link);
+    })
+    .catch((err) => {
+      console.error(`Avatar not updated: `, err); // log the error to the console
```

Comment:
Good job that you `catch` possible errors at the end of server requests. You can simplify it with `.catch(console.error);` because `console.error` is meant to log errors